### ISSUE TITLE
what to set for input_seq and target_seq for train-data -- asked by a beginner in NLP domain :-)

### ISSUE THREAD

**Issue by Hengameh400** [JDAI-CV/image-captioning#21]
I want to use your sent-decoder and generate captions over my 2D images.
for each image, i have a ground-truth caption.

i will encode each word of my captions using *coco_vocabulary.txt*  to an integer; and if i am not mistaken, this is what you did over your captions and you saved encoded captions to  target_seq for train-data at pkl files
https://github.com/JDAI-CV/image-captioning/blob/master/datasets/coco_dataset.py#L27

my understanding is that target_seq for train-data are dictionaries
keys represent image-ids
values represent encoded captions

am i right?

looking forward to hearing from you!

Thanks!

**Comment by Hengameh400** [comment]
also, could you please kindly guide me what is the difference between:
self.rl_stage at https://github.com/JDAI-CV/image-captioning/blob/master/main.py#L195
to be False or True?

**Comment by YehLi** [comment]
When self.rl_stage == False, the model is trained by cross entropy loss. If self.rl_stage == True, the model is optimized by reinforcement learning. To create the pkl files, you can refer to https://github.com/ruotianluo/ImageCaptioning.pytorch/blob/master/scripts/prepro_labels.py
